proposals/LP0004-project-governance.md

```diff
@@ -0,0 +1,393 @@
+
+_Area teams_ are not intended to be direction setters or primary maintainers of their areas, although individuals on an _area team_ may fulfill that role separately. The _area team’s_ role is as a steward and moderator ensuring the health and smooth operation of the area.
+
+Each _area team_ will have 5 members elected by the _voting contributors_. Candidates for _area teams_ must be a _voting contributor_ and self-nominated. An individual cannot serve on two _area teams_.
+
+Members of an _area team_ are elected for 1 year terms. Individuals are encouraged to not run for more than two consecutive terms. To encourage turnover, incumbent members of an _area team_ must place in the top 4 candidates during an election. The fifth spot will always go to a non-incumbent candidate.
+
+
+#### Initial Area Teams
+
+With this proposal a group of initial _area teams_ will be formed with defined domain areas. The members of each _area team_ will be defined by an election held after the adoption of the proposal. The initial _area teams_ proposed below are based on statistics generated from the LLVM project repository (See Appendix 2 for applicable scripts).
+
+In dividing the LLVM project into _area teams_, this proposal seeks to break up areas of the project based on contribution volume and number of active contributors. This proposal assumes that areas of the project with more unique contributors are more likely to have contributor disagreements, and that areas with more highly active contributors will be more likely to have diverse _area team_ formations.
+
+This proposal puts a possible set of _area teams_ to form with the specified areas of ownership. One notable point is that LLVM (`:/llvm`) had 945 unique authors contribute over the 12 month period used for statistics. This is far more than any other part of the project, and it justified splitting LLVM into two teams, which are separated as the _LLVM Backend_ and _LLVM Middle-end_. Clang had 706 unique authors, which is also significantly higher than other sub-projects. To represent different parts of the Clang community the _Clang_ and _Clang Tooling_ teams are separated below.
+
+* LLVM Backend (CodeGen, MC, Target)
+* LLVM Middle-end (Everything not covered by LLVM Backend)
+* Clang (All the `clang` sub-project except tooling and format libraries and tools)
+* Clang Tooling (`clang-tools-extra` and the tooling and format libraries from `clang`)
+* C/C++ Runtime Libraries (`libcxx`, `libc`, `libunwind`, `libcxxabi`)
+* Compiler Runtimes (`compiler-rt`, `openmp`)
+* Flang (`flang`)
+* MLIR (`mlir`)
+* Binary tools (`lldb`, `lld`, `bolt`)
```

Comment:
How were these area teams defined? For example, why does Flang have its own area time, while LLDB is part of Binary tools, despite the two projects having roughly the same number of commits over the last year?

Reply:
Intuitively to me, LLDB should be its own thing considering it's a big project with a lot of moving parts that doesn't have a ton in common with other LLVM tools. It's an _interactive environment_ that allows programmers to _infer information about programs_. The other tools listed _consume_ and _produce_ binaries.

IIUC we want to technologies (in-part) based off the amount of technical overlap contributors to those technologies probably have. E.g., people working on "runtimey stuff" probably have some sort of base "common-sense" knowledge about "runtimey things" that folks working on like, I dunno, FileCheck, aren't guaranteed to have. I can't see a ton of overlap between, say, BOLT and LLDB right now. (LLD and BOLT being together makes sense though IMO.)

So, therefore, I think that if we're going to create an umbrella for LLDB, I think it should be something like "debugging tools" or "interactive tools." Or, we don't make an umbrella and we just say LLDB is its own thing.